src/client/scripts/esm/chess/variants/gamerules.js

```diff
@@ -85,7 +87,7 @@ function GameRules() {
      * An object containing arrays of types white and black can promote to, if it's legal for them to promote.
      * If one color can't promote, their list should be left undefined.
      * If no color can promote, this should be left undefined.
-	 * @type {{ white: string[], black: string[] }}
+	 * @type {Object<string, string[]>}
```

Comment:
`Object<string, string[]> | undefined`
as it's not required, and won't be present if `promotionRanks` is also undefined.